app/models/concerns/solution_base.rb

```diff
@@ -61,6 +61,18 @@ def latest_exercise_version?
     true
   end
 
+  def team_solution?
```

Comment:
I'm a bit scared of this and what this would imply in the future. I'm okay with this right now, but asking the solution what kind of solution it is doesn't seem like good design as it's the same as asking what class the object is.

Let's see where we get to and refactor later.

Reply:
Yeah, agree. That method is already used throughout the codebase though. This is just a DRYing up of the existing code to move it into one place.